atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -219,40 +221,101 @@ def _get_remote_urls_helper(self, koji_build):
+                if archive['filename'] == REMOTE_SOURCES_FILENAME:
+                    archive_found = True
+                elif archive['filename'] == REMOTE_SOURCES_JSON_FILENAME:
+                    json_found = True
+                all_archives.append(archive['filename'])
+
+            if not (archive_found and json_found):
+                message = ''
+                if not archive_found:
+                    message = 'Remote sources archive'
+                if not json_found:
+                    if message:
+                        message += ' and remote sources json'
+                    else:
+                        message = 'Remote sources json'
+                message += ' missing.'
+                raise RuntimeError(message)
+
+            elif len(archives) > 2:
+                raise RuntimeError('There can be just one remote sources archive and one '
+                                   'remote sources json, got: {}'.format(all_archives))
+
+        elif 'remote_sources' in koji_build['extra']['image']:
+            self.log.debug('remote_sources defined')
+            remote_sources = koji_build['extra']['typeinfo']['remote-sources']
+            wrong_archives = False
+            all_archives = set()
+
+            for remote_s in remote_sources:
+                remote_archive = None
+                remote_json = None
+
+                if len(remote_s['archives']) != 2:
+                    self.log.error('remote source "%s" does not contain 2 archives, but "%s"',
+                                   remote_s['name'], remote_s['archives'])
+                    wrong_archives = True
+                else:
+                    for archive in remote_s['archives']:
+                        if archive.endswith('.json'):
+                            remote_json = archive
+                        else:
+                            remote_archive = archive
+
+                    if not remote_json:
+                        self.log.error('remote source json, for remote source "%s" not found '
+                                       'in archives "%s"', remote_s['name'], remote_s['archives'])
+                        wrong_archives = True
+                    else:
+                        remote_source = {}
+                        remote_source['url'] = os.path.join(remote_sources_path, remote_archive)
+                        remote_source['dest'] = '-'.join([koji_build['nvr'], remote_archive])
+                        remote_sources_urls.append(remote_source)
+                        cachito_json_url = os.path.join(remote_sources_path, remote_json)
+                        remote_json_map[remote_source['dest']] = cachito_json_url
+                        all_archives.add(remote_archive)
+                        all_archives.add(remote_json)
+
+            if wrong_archives:
+                raise RuntimeError('Problems with archives in remote sources: {}'.
+                                   format(remote_sources))
+
```

Comment:
Can be this extracted into separate method for better readability?

Reply:
ok